config/routes.rb

```diff
@@ -446,6 +446,7 @@
   resources :main, only: [] do
     collection do
       post 'logout'
+      get 'settings'
```

Comment:
I'd like an opinion if placing the route here makes sense. Most of the functionality will be provided through the REST API, meaning I was unsure whether to put it here (since this feature is very general), in the api namespace, or in it's own resource entirely.

Reply:
@anoverga17 it's more standard to use a separate `admin/` namespace. See https://guides.rubyonrails.org/routing.html#controller-namespaces-and-routing.

So we'd have `admin/courses/`, `admin/courses/1/edit`, etc.